src/CallbackSerializer.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * This file is part of the prooph/pdo-snapshot-store.
+ * (c) 2016-2017 prooph software GmbH <contact@prooph.de>
+ * (c) 2016-2017 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\SnapshotStore;
+
+final class CallbackSerializer implements Serializer
+{
+    /**
+     * callable
+     */
+    private $serializeCallback = 'serialize';
+
+    /**
+     * callable
+     */
+    private $unserializeCallback = 'unserialize';
+
+    public function __construct(?callable $serializeCallback, ?callable $unserializeCallback)
+    {
+        if (null !== $serializeCallback && null !== $unserializeCallback) {
```

Comment:
I am against changing timestamp or anything else within the snapshot. If you want to do some crazy weird stuff, implement your own snapshot-store.